src/actions/register_user.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module to interact with Register User action."""
+
+import logging
+
+import ops
+
+# pydantic is causing this no-name-in-module problem
+from pydantic import ValidationError  # pylint: disable=no-name-in-module,import-error
+
+import synapse
+from user import User
+
+logger = logging.getLogger(__name__)
+
+
+class RegisterUserError(Exception):
+    """Exception raised when something fails while running register-user.
+
+    Attrs:
+        msg (str): Explanation of the error.
+    """
+
+    def __init__(self, msg: str):
+        """Initialize a new instance of the RegisterUserError exception.
+
+        Args:
+            msg (str): Explanation of the error.
+        """
+        self.msg = msg
+
+
+# access_token is not a password
+def register_user(
+    container: ops.Container,
+    username: str,
+    admin: bool,
+    server: str = "",
+    admin_access_token: str = "",  # nosec
```

Comment:
A 'f"User {user.username} exists but there is no server/admin access token set."' will be raised. The token is used in case the user already exists so it will return the existing access token.

Reply:
Related: #52